packages/backend/src/controllers/notificationsController.ts

```diff
@@ -0,0 +1,107 @@
+import {
+    ApiCreateNotification,
+    ApiErrorPayload,
+    ApiGetNotifications,
+    NotificationType,
+} from '@lightdash/common';
+import {
+    Controller,
+    Get,
+    Middlewares,
+    OperationId,
+    Patch,
+    Path,
+    Post,
+    Query,
+    Request,
+    Response,
+    Route,
+    SuccessResponse,
+    Tags,
+} from '@tsoa/runtime';
+import express from 'express';
+import { notificationsService } from '../services/services';
+import { allowApiKeyAuthentication, isAuthenticated } from './authentication';
+
+@Route('/api/v1/notifications')
+@Response<ApiErrorPayload>('default', 'Error')
+@Tags('Notifications')
+export class NotificationsController extends Controller {
+    /**
+     * Gets notifications for a user based on the type
+     * @param req express request
+     * @query type the type of notification to get
+     * @returns the notifications for a user
+     */
+    @Middlewares([allowApiKeyAuthentication, isAuthenticated])
+    @SuccessResponse('200', 'Success')
+    @Get('/')
+    @OperationId('getNotifications')
+    async getNotifications(
+        @Request() req: express.Request,
+        @Query() type: NotificationType,
+    ): Promise<ApiGetNotifications> {
+        let results: ApiGetNotifications['results'] = [];
+        switch (type) {
+            case NotificationType.DASHBOARD_COMMENTS:
+                results =
+                    await notificationsService.getDashboardCommentNotifications(
+                        req.user!.userUuid,
+                    );
+                break;
+            default:
+                break;
+        }
+        this.setStatus(200);
+        return {
+            status: 'ok',
+            results,
+        };
+    }
+
+    /**
+     * Create notification for a dashboard comment
+     * @param req express request
+     * @param dashboardUuid the uuid of the dashboard
+     */
+    @Middlewares([allowApiKeyAuthentication, isAuthenticated])
+    @SuccessResponse('200', 'Success')
+    @Post('{commentId}/dashboard/{dashboardUuid}')
+    @OperationId('createNotificationForDashboardComment')
+    async createDashboardCommentNotification(
```

Comment:
Do we want to trigger this on the frontend (with react-query) once the comment is successfully created, or should we call this `notificationsService.createDashboardCommentNotification` in `CommentsController`? 🤔

Reply:
I prefer to call this notificationsService.createDashboardCommentNotification in  createNotification in the service.